packages/flutter/test/widgets/binding_test.dart

```diff
@@ -46,22 +46,18 @@ class PushRouteInformationObserver with WidgetsBindingObserver {
 }
 
 void main() {
-  setUp(() {
-    WidgetsFlutterBinding.ensureInitialized();
-  });
-
   Future<void> setAppLifeCycleState(AppLifecycleState state) async {
     final ByteData? message =
         const StringCodec().encodeMessage(state.toString());
-    await ServicesBinding.instance.defaultBinaryMessenger
-        .handlePlatformMessage('flutter/lifecycle', message, (_) {});
+    await TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+        .handlePlatformMessage('flutter/lifecycle', message, (_) { });
```

Comment:
It looks like this method would have access to `tester` in all the places it's invoked as well. Can we pass a `WidgetTester` or the binding/messenger directly to this method?

Reply:
Same as above: It's the same thing, just more work producing a larger diff.